packages/terra-tag/src/Tag.jsx

```diff
@@ -42,30 +41,29 @@ const propTypes = {
 class Tag extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { focused: false };
     this.handleKeyUp = this.handleKeyUp.bind(this);
     this.handleOnBlur = this.handleOnBlur.bind(this);
+    this.handleClick = this.handleClick.bind(this);
   }
 
   handleOnBlur(event) {
-    this.setState({ focused: false });
-
     if (this.props.onBlur) {
       this.props.onBlur(event);
     }
   }
 
   handleKeyUp(event) {
-    // Apply focus styles for keyboard navigation
-    if (event.nativeEvent.keyCode === KeyCode.KEY_TAB) {
-      this.setState({ focused: true });
-    }
-
     if (this.props.onKeyUp) {
       this.props.onKeyUp(event);
     }
   }
 
+  handleClick(event) {
+    if (this.props.onClick) {
+      this.props.onClick(event);
+    }
+  }
```

Comment:
Do you not have a requirement for this to also be triggered Return and/or Space?

Reply:
Terra-Tag can be either button or links based on props provided.  "handleClick" handler gets invoked on both space and return key for button by default. For hyperlink "handleClick" handler will be triggered by only return key.